packages/graphback-runtime-knex/src/KnexDBDataProvider.ts

```diff
@@ -16,16 +16,17 @@ import { buildQuery } from './knexQueryMapper';
  */
 //tslint:disable-next-line: no-any
 export class KnexDBDataProvider<Type = any, GraphbackContext = any> implements GraphbackDataProvider<Type, GraphbackContext>{
-
   protected db: Knex;
   protected baseType: GraphQLObjectType;
   protected tableName: string;
   protected tableMap: ModelTableMap;
 
-  public constructor(baseType: GraphQLObjectType, db: Knex) {
-    this.db = db;
-    this.baseType = baseType;
-    //TODO build and use mapping here
+  public constructor(dbConfig: Knex.Config) {
+    this.db = Knex(dbConfig);
```

Comment:
Trimmed this down to only require the database config. If we think about the user adding their own custom default crud service, it would not work because the constructor values would be different for every model.

See API:
```
const services = buildGraphbackServices({
  defaultDataProvider: new PgKnexDBDataProvider(dbConfig),
  pubSub
})
```

If the model was required here, this simply would not work.

Reply:
Yes. This makes PgKnexDBDataProvider stateless so service itself will hold some state for types. I really like how you figured that out. 

What service this is going to use? 
For example for upstream you are using CRUDService, For datasync it will use DataSyncService (as compositve of Auth, Metrics and CRUD service). 

Are we simplifying things only for the upstream and going to have some hacks for the other use cases? (Which is fine tradeoff)

Reply:
The constructor can still accept whatever generic information it needs for configuration, but this approach will fail if someone wants to add some model specific info to the CRUD service in the constructor.

But they can't really do this now anyway, since they don't have that information?